contracts/StorageAccessible.sol

```diff
@@ -52,20 +52,47 @@ contract StorageAccessible {
      */
     function simulateDelegatecall(
         address targetContract,
-        bytes memory calldataPayload
+        bytes calldata calldataPayload
     ) public returns (bytes memory response) {
-        bytes memory innerCall = abi.encodeWithSelector(
-            this.simulateDelegatecallInternal.selector,
-            targetContract,
-            calldataPayload
-        );
-        (, response) = address(this).call(innerCall);
-        bool innerSuccess = response[response.length - 1] == 0x01;
-        setLength(response, response.length - 1);
-        if (innerSuccess) {
-            return response;
-        } else {
-            revertWith(response);
+        bytes4 selector = this.simulateDelegatecallInternal.selector;
+        assembly {
+            let internalCalldata := mload(0x40)
+            mstore(internalCalldata, selector)
+            // Abuse the fact that both this and the internal methods have the
+            // same signature, and differ only in symbol name (and therefore,
+            // selector) and copy calldata directly. This saves us approximately
+            // 250 bytes of code and 300 gas at runtime over the
+            // `abi.encodeWithSelector` builtin.
+            calldatacopy(
+                add(internalCalldata, 0x04),
+                0x04,
+                sub(calldatasize(), 0x04)
+            )
+
+            pop(call(
```

Comment:
Would love a comment on the `pop`

Reply:
I assume this is primarily to free up stack?

Reply:
It was required by the compiler. If you omit it you get an error:
```
TypeError: Top-level expressions are not supposed to return values (this expression returns 1 value). Use ``pop()`
` or assign them.
            call(
            ^ (Relevant source part starts here and spans across multiple lines).
```

I'll add a comment why its necessary.